packages/react-components/react-tree/src/components/FlatTree/useFlatTree.ts

```diff
@@ -11,12 +11,25 @@ import { useTreeContext_unstable } from '../../contexts/treeContext';
 import { useSubtree } from '../../hooks/useSubtree';
 import { ImmutableSet } from '../../utils/ImmutableSet';
 import { ImmutableMap } from '../../utils/ImmutableMap';
+import { useSubtreeContext_unstable } from '../../contexts/subtreeContext';
 
 export const useFlatTree_unstable: (props: FlatTreeProps, ref: React.Ref<HTMLElement>) => FlatTreeState = (
   props,
   ref,
 ) => {
-  const level = useTreeContext_unstable(ctx => ctx.level);
+  const { level } = useSubtreeContext_unstable();
+  if (
+    useTreeContext_unstable(ctx => ctx.treeType) === 'nested' &&
```

Comment:
If you need to check if a parent context is defined or not you can try use this hook https://github.com/microsoft/fluentui/blob/aed130bfbf4ac88947a697b723387fd8d8382686/packages/react-components/react-context-selector/src/useHasParentContext.ts#L12-L20